File: examples/python/signal_example.py
"""
Surmado Signal - AI Visibility Testing Example
Test how your brand appears across 7 AI platforms:
ChatGPT, Perplexity, Google Gemini, Claude, Meta AI, Grok, DeepSeek
Docs: https://help.surmado.com/docs/api-reference/
"""
import os
import time
import requests

API_KEY = os.environ.get("SURMADO_API_KEY", "YOUR_API_KEY")
BASE_URL = "https://api.surmado.com/v1"


def create_signal_report(
    url: str,
    brand_name: str,
    email: str,
    industry: str,
    location: str,
    persona: str,
    pain_points: str,
    brand_details: str,
    direct_competitors: str,
    tier: str = "basic",
):
    """
    Create a Signal (AI visibility) report.
    
    Args:
        url: Your website URL
        brand_name: Your brand name
        email: Email for notifications
        industry: Your industry/sector
        location: Primary market location
        persona: Target customer description
        pain_points: Problems your product solves (string, not list)
        brand_details: Your brand positioning
        direct_competitors: Comma-separated competitor names (string, not list)
        tier: "basic" (1 credit, $25) or "pro" (2 credits, $50)
    
    Returns:
        Report data with report_id
    """
    response = requests.post(
        f"{BASE_URL}/reports/signal",
        headers={
            "X-API-Key": API_KEY,
            "Content-Type": "application/json",
        },
        json={
            "url": url,
            "brand_name": brand_name,
            "email": email,
            "industry": industry,
            "location": location,
            "persona": persona,
            "pain_points": pain_points,  # Must be a string
            "brand_details": brand_details,
            "direct_competitors": direct_competitors,  # Must be a string
            "tier": tier,
        },
    )
    response.raise_for_status()
    return response.json()


def get_report(report_id: str):
    """Get report status and results."""
    response = requests.get(
        f"{BASE_URL}/reports/{report_id}",
        headers={"X-API-Key": API_KEY},
    )
    response.raise_for_status()
    return response.json()


def wait_for_completion(report_id: str, timeout_minutes: int = 20):
    """Poll until report completes or fails."""
    start = time.time()
    timeout_seconds = timeout_minutes * 60
    
    while time.time() - start < timeout_seconds:
        report = get_report(report_id)
        status = report.get("status")
        
        if status == "completed":
            print(f"Report completed in {int(time.time() - start)} seconds")
            return report
        elif status == "failed":
            raise Exception(f"Report failed: {report.get('error')}")
        
        print(f"Status: {status}. Waiting...")
        time.sleep(30)
    
    raise TimeoutError(f"Report did not complete within {timeout_minutes} minutes")


if __name__ == "__main__":
    # Create a Signal report
    print("Creating Signal report...")
    result = create_signal_report(
        url="https://example.com",
        brand_name="Example Brand",
        email="you@example.com",
        industry="E-commerce",
        location="United States",
        persona="Small business owners looking for affordable solutions",
        pain_points="Finding reliable vendors, managing inventory costs",
        brand_details="We provide affordable e-commerce solutions for growing businesses",
        direct_competitors="Shopify, BigCommerce, WooCommerce",
        tier="basic",
    )
    
    report_id = result["report_id"]
    print(f"Report created: {report_id}")
    print(f"Credits used: {result.get('credits_used')}")
    
    # Wait for completion
    print("\nWaiting for report to complete...")
    completed = wait_for_completion(report_id)
    
    # Download results
    if completed.get("download_url"):
        print(f"\nPDF: {completed['download_url']}")
    if completed.get("pptx_download_url"):
        print(f"PPTX: {completed['pptx_download_url']}")
    if completed.get("intelligence_download_url"):
        print(f"JSON: {completed['intelligence_download_url']}")

File: examples/javascript/signal_example.js
/**
 * Surmado Signal - AI Visibility Testing Example
 * Test how your brand appears across 7 AI platforms:
 * ChatGPT, Perplexity, Google Gemini, Claude, Meta AI, Grok, DeepSeek
 * Docs: https://help.surmado.com/docs/api-reference/
 */

const API_KEY = process.env.SURMADO_API_KEY || 'YOUR_API_KEY';
const BASE_URL = 'https://api.surmado.com/v1';

/**
 * Create a Signal (AI visibility) report.
 * @param {Object} params - Report parameters
 * @returns {Promise<Object>} Report data with report_id
 */
async function createSignalReport({
  url,
  brandName,
  email,
  industry,
  location,
  persona,
  painPoints,
  brandDetails,
  directCompetitors,
  tier = 'basic',
}) {
  const response = await fetch(`${BASE_URL}/reports/signal`, {
    method: 'POST',
    headers: {
      'X-API-Key': API_KEY,
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({
      url,
      brand_name: brandName,
      email,
      industry,
      location,
      persona,
      pain_points: painPoints, // Must be a string, not an array
      brand_details: brandDetails,
      direct_competitors: directCompetitors, // Must be a string, not an array
      tier,
    }),
  });

  if (!response.ok) {
    const error = await response.json();
    throw new Error(`API error: ${JSON.stringify(error)}`);
  }

  return response.json();
}

/**
 * Get report status and results.
 */
async function getReport(reportId) {
  const response = await fetch(`${BASE_URL}/reports/${reportId}`, {
    headers: { 'X-API-Key': API_KEY },
  });

  if (!response.ok) {
    const error = await response.json();
    throw new Error(`API error: ${JSON.stringify(error)}`);
  }

  return response.json();
}

/**
 * Poll until report completes or fails.
 */
async function waitForCompletion(reportId, timeoutMinutes = 20) {
  const start = Date.now();
  const timeoutMs = timeoutMinutes * 60 * 1000;

  while (Date.now() - start < timeoutMs) {
    const report = await getReport(reportId);
    const status = report.status;

    if (status === 'completed') {
      console.log(`Report completed in ${Math.round((Date.now() - start) / 1000)} seconds`);
      return report;
    } else if (status === 'failed') {
      throw new Error(`Report failed: ${report.error}`);
    }

    console.log(`Status: ${status}. Waiting...`);
    await new Promise(resolve => setTimeout(resolve, 30000));
  }

  throw new Error(`Report did not complete within ${timeoutMinutes} minutes`);
}

// Main example
async function main() {
  try {
    // Create a Signal report
    console.log('Creating Signal report...');
    const result = await createSignalReport({
      url: 'https://example.com',
      brandName: 'Example Brand',
      email: 'you@example.com',
      industry: 'E-commerce',
      location: 'United States',
      persona: 'Small business owners looking for affordable solutions',
      painPoints: 'Finding reliable vendors, managing inventory costs',
      brandDetails: 'We provide affordable e-commerce solutions for growing businesses',
      directCompetitors: 'Shopify, BigCommerce, WooCommerce',
      tier: 'basic',
    });

    const reportId = result.report_id;
    console.log(`Report created: ${reportId}`);
    console.log(`Credits used: ${result.credits_used}`);

    // Wait for completion
    console.log('\nWaiting for report to complete...');
    const completed = await waitForCompletion(reportId);

    // Download results
    if (completed.download_url) {
      console.log(`\nPDF: ${completed.download_url}`);
    }
    if (completed.pptx_download_url) {
      console.log(`PPTX: ${completed.pptx_download_url}`);
    }
    if (completed.intelligence_download_url) {
      console.log(`JSON: ${completed.intelligence_download_url}`);
    }
  } catch (error) {
    console.error('Error:', error.message);
    process.exit(1);
  }
}

main();

